src/commands/crashlytics-symbols-upload.ts

```diff
@@ -0,0 +1,195 @@
+import * as fs from "fs-extra";
+import * as os from "os";
+import * as path from "path";
+import * as spawn from "cross-spawn";
+import * as uuid from "uuid";
+
+import { Command } from "../command";
+import { FirebaseError } from "../error";
+import * as utils from "../utils";
+import * as downloadUtils from "../downloadUtils";
+
+enum SymbolGenerator {
+  breakpad = "breakpad",
+  csym = "csym",
+}
+
+interface Options {
+  app: string | null;
+  generator: SymbolGenerator | null;
+  dryRun: boolean | null;
+  debug: boolean | null;
+  // Temporary override to use a local JAR until we get the fat jar in our
+  // bucket
+  localJar: string | null;
+}
+
+interface JarOptions {
+  jarFile: string;
+  app: string;
+  generator: SymbolGenerator;
+  cachePath: string;
+  symbolFile: string;
+  generate: boolean;
+}
+
+const SYMBOL_CACHE_ROOT_DIR = process.env.FIREBASE_CRASHLYTICS_CACHE_PATH || os.tmpdir();
+const JAR_CACHE_DIR =
+  process.env.FIREBASE_CRASHLYTICS_BUILDTOOLS_PATH ||
+  path.join(os.homedir(), ".cache", "firebase", "crashlytics", "buildtools");
+const JAR_VERSION = "2.7.1";
+const JAR_URL = `https://dl.google.com/android/maven2/com/google/firebase/firebase-crashlytics-buildtools/${JAR_VERSION}/firebase-crashlytics-buildtools-${JAR_VERSION}.jar`;
+
+export default new Command("crashlytics:symbols:upload <symbol-files...>")
+  .description("Upload symbols for native code, to symbolicate stack traces.")
+  .option("--app <app_id>", "the app id of your Firebase app")
+  .option("--generator [breakpad|csym]", "the symbol generator being used, defaults to breakpad.")
+  .option("--dry-run", "generate symbols without uploading them")
+  .option("--debug", "print debug output and logging from the underlying uploader tool")
+  .option("--local-jar <path>", "override the fetched jar with one on the filesystem")
```

Comment:
Is there a way to make hidden flags? It didn't seem to accept the option unless I added this

Reply:
Maybe I can use an environment variable instead